src/bin/utils/schema.ts

```diff
@@ -2,16 +2,46 @@ import fs from 'fs/promises';
 import path from 'path';
 import ts from 'typescript';
 
+import { exists } from '@/src/bin/utils/file';
+import type { Schema } from '@/src/types/schema';
 import { generateUniqueId } from '@/src/utils/id';
 
-export async function parseSchemaDefinitionFile(filePath: string = './schemas/index.d.ts') {
+const DEFAULT_SCHEMA_SUMMARY = 'This is a schema summary';
+
+/**
+ * Generates a unique field ID.
+ *
+ * @param type The type of the field.
+ */
+export const generateFieldId = (type: string): string => {
+  return `${type}-${generateUniqueId()}`;
+};
+
+/**
+ * Parses the schema definition file and returns an array of Schema objects.
```

Comment:
```suggestion
 * Parses the schema definition file and returns an array of `Schema` objects.
```

Code should be in backticks.